src/tic_tac_toe/referee.clj

```diff
@@ -1,56 +1,23 @@
 (ns tic-tac-toe.referee
-  (:require [tic-tac-toe.board :as board])
+  (:require [tic-tac-toe.board :as board]
+            [tic-tac-toe.user-interface :as ui]
+            [tic-tac-toe.validation.board-validator :refer :all]
+            [tic-tac-toe.validation.move-validator :refer :all])
   (:use [clojure.core.match :refer [match]]))
 
-(defn convert-to-spot
-  [move]
-  (if (string? move)
-    (- (Integer/parseInt move) 1)
-    move))
-
 (defn validate-move
-  [player move]
-  (convert-to-spot move))
+  [request-move game-ui player board move]
+  (try
+    (check-spot board move)
+    (catch Throwable t
+      (->> (.getMessage t)
+           (partial ui/prompt-move game-ui board player)
+           (request-move game-ui player board)))))
```

Comment:
Although it's only one level deep here, sometimes functions that throw can be 4 or 5 calls removed, making it less obvious what's throwing and why. Try to only use exception handling in truly exceptional cases and avoid error handling as control flow.

Reply:
OK, [refactored](https://github.com/saralein/tic-tac-toe_clojure/pull/8/files#diff-6f8e1408e5b8dfe720ab43fcbf017b8fR12) in this area so nothing throws.  Figured out how to check with re-matches so even the string never reaches a point where it might throw.